src/target/riscv/riscv-013.c

```diff
@@ -3082,36 +3082,46 @@ static int read_sbcs_nonbusy(struct target *target, uint32_t *sbcs)
 	}
 }
 
+/* TODO: return mem_access_result_t */
 static int modify_privilege(struct target *target, uint64_t *mstatus, uint64_t *mstatus_old)
 {
-	if (riscv_virt2phys_mode_is_hw(target)
-			&& has_sufficient_progbuf(target, 5)) {
-		/* Read DCSR */
-		uint64_t dcsr;
-		if (register_read_direct(target, &dcsr, GDB_REGNO_DCSR) != ERROR_OK)
-			return ERROR_FAIL;
+	if (!riscv_virt2phys_mode_is_hw(target))
```

Comment:
Detail, if you like:

```suggestion
	assert(mstatus);
	assert(mstatus_old);

	if (!riscv_virt2phys_mode_is_hw(target))
```

Reply:
Addressed